test/Lombiq.OSOCE.Tests.UI/Tests/ModuleTests/BehaviorMaintenanceTests.cs

```diff
@@ -24,9 +27,27 @@ public BehaviorMaintenanceTests(ITestOutputHelper testOutputHelper)
             context => context.TestSiteOwnerPermissionToRoleMaintenanceExecutionAsync(),
             configuration => configuration.SetAddSiteOwnerPermissionToRoleMaintenanceConfiguration());
 
+    // This test employs ExecuteTestAsync with a distinct setup delegate instead of ExecuteTestAfterSetupAsync.
+    // This choice is prompted by the modifications it applies to the appsettings.json file, occasionally leading
+    // to failures in subsequent tests within the NuGet solution in the CI environment
     [Fact]
     public Task ChangeUserSensitiveContentMaintenanceTaskShouldBeExecutedSuccessfully() =>
-        ExecuteTestAfterSetupAsync(
+        ExecuteTestAsync(
             context => context.ChangeUserSensitiveContentMaintenanceExecutionAsync(),
+            async context =>
+            {
+                var homepageUri = await context.GoToSetupPageAndSetupOrchardCoreAsync(
+                    new OrchardCoreSetupParameters(context)
+                    {
+                        SiteName = "Lombiq's OSOCE - UI Testing",
+                        RecipeId = "Lombiq.OSOCE.Tests",
+                        TablePrefix = "OSOCE",
+                        SiteTimeZoneValue = "Europe/Budapest",
+                    });
+
+                context.Exists(By.Id("navbar"));
+
+                return homepageUri;
```

Comment:
UITT determines if two setup delegates are the same like this: https://github.com/Lombiq/UI-Testing-Toolbox/blob/88ec8caf6940e7f1445fe4993701fc0ce5ce5e48/Lombiq.Tests.UI/Services/UITestExecutionSession.cs#L646 You can trick this without copying over this much code by only running `SetupHelpers.RunSetupAsync()` in this delegate like this I think:

```suggestion
                var homepageUri = await Lombiq.Tests.UI.Samples.Helpers.SetupHelpers.RunSetupAsync(context);
                return homepageUri;
```

But please check. Perhaps adding something little, like another `context` call after `SetupHelpers.RunSetupAsync()` would also be needed.

Reply:
Do we need this change in the Nuget aswell of that could remain?

Reply:
Also do we intentionally want to get rid of `context.Exists(By.id....)` ?

Reply:
NuGet solution too, of course.

The exists check is needed in `RunSetupAsync()` to check for the success of the setup.